opal/mca/btl/ugni/btl_ugni.h

```diff
@@ -303,10 +303,10 @@ OPAL_MODULE_DECLSPEC extern mca_btl_ugni_module_t mca_btl_ugni_module;
 
 static inline uint32_t mca_btl_ugni_ep_get_device_index (mca_btl_ugni_module_t *ugni_module)
 {
-    static volatile uint32_t device_index = (uint32_t) 0;
+    static opal_atomic_int64_t device_index = (uint32_t) 0;
```

Comment:
How many OFI devices are we expecting here ? Maybe you want to cast the right hand side to uint64_t as well.

Reply:
This index will be incremented by the number of unique threads that use RMA. That is very unlikely to be as large as 2^32-1 but figured it would be better to just make it a uint64_t.

There will only ever be at most 128 uGNI contexts.